packages/backend/src/api/controllers/tvl/TvlController.ts

```diff
@@ -100,6 +115,62 @@ export class TvlController {
     }
   }
 
+  async getAggregatedApiResponse(slugs: string[]): Promise<TvlProjectResult> {
+    const dataTimings = await this.getDataTimings()
+
+    const projectIdsFilter = [...layer2s, ...bridges]
+      .filter((project) => slugs.includes(project.display.slug))
+      .map((project) => project.id)
+
+    if (!dataTimings.latestTimestamp) {
+      return {
+        result: 'error',
+        error: 'NO_DATA',
+      }
+    }
+
+    if (!dataTimings.isSynced && this.options.errorOnUnsyncedTvl) {
+      return {
+        result: 'error',
+        error: 'DATA_NOT_FULLY_SYNCED',
+      }
+    }
+
+    const [hourlyReports, sixHourlyReports, dailyReports] = await Promise.all([
+      this.aggregatedReportRepository.getHourly(
+        getHourlyMinTimestamp(dataTimings.latestTimestamp),
+        'TVL',
+      ),
+      this.aggregatedReportRepository.getSixHourly(
+        getSixHourlyMinTimestamp(dataTimings.latestTimestamp),
+        'TVL',
+      ),
+      this.aggregatedReportRepository.getDaily('TVL'),
+    ])
+
+    const projectIdsFilterSet = new Set(
+      projectIdsFilter.map((x) => x.toString()),
+    )
+
+    const tvlApiProjectResponse = generateAggregatedApiResponse(
+      hourlyReports,
+      sixHourlyReports,
+      dailyReports,
+      this.projects
```

Comment:
here you can just pass the projectIdsFilter

Reply:
hourly, sixHourly and daily report values are needed inside `generateAggregatedApiResponse`